x/ecocredit/server/core/query_batches_by_class.go

```diff
@@ -0,0 +1,46 @@
+package core
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+// BatchesByClass queries all batches issued under a given credit class.
+func (k Keeper) BatchesByClass(ctx context.Context, request *core.QueryBatchesByClassRequest) (*core.QueryBatchesByClassResponse, error) {
+	pg, err := GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+	class, err := k.stateStore.ClassInfoStore().GetByName(ctx, request.ClassId)
+	if err != nil {
+		return nil, err
+	}
+	// we put a "-" after the class name to avoid including class names outside of the query (i.e. a query for C01 could technically include C011 otherwise).
+	it, err := k.stateStore.BatchInfoStore().List(ctx, api.BatchInfoBatchDenomIndexKey{}.WithBatchDenom(class.Name+"-"), ormlist.Paginate(pg))
```

Comment:
I was able to successfully run the tests without "-" added here. Are you sure this is necessary?

Reply:
It's not that queries won't run successfully, its that it could include classes you weren't necessarily querying for.

Like in the example in the comment, if there are two classes, "C01" and "C011" (for whatever reason), without the added "-", it would include "C011" or any other class that starts with C01. the "-" ensures we _only_ get C01 batches

Reply:
added a test case to highlight this, if you remove the dash you'll see it fails without it